src/eventProcessor/GoogleAnalyticsEventProcessor.js

```diff
@@ -0,0 +1,129 @@
+goog.module('measurementLibrary.eventProcessor.GoogleAnalyticsEventProcessor');
+
+/**
+ * A class that processes events pushed to the data layer
+ * by constructing and sending Google Analytics events via
+ * Measurement Protocol.
+ * For its Google Analytic events, it generates a unique client ID
+ * and stores it in the long term storage model provided.
+ * @implements {EventProcessor}
+ */
+class GoogleAnalyticsEventProcessor {
+  /**
+   * @param {{
+   *     'api_secret': (string|undefined),
+   *     'measurement_id': (string|undefined),
+   *     'measurement_url': string,
+   *     'client_id_expires': number,
+   *     'automatic_params': !Object<string, boolean>,
+   * }=} optionsObject
+   */
+  constructor({
+      'api_secret': apiSecret,
+      'measurement_id': measurementId,
+      'measurement_url': measurementUrl = 'https://www.google-analytics.com/mp/collect',
```

Comment:
Could be worth pulling out to a top-level constant, e.g. DEFAULT_MEASUREMENT_URL (especially since it sounds like this will have two possible values depending on compiler flags in a later PR).